### PR TITLE
Update element templates docs

### LINKED ISSUE #2723: Update element template docs
__What should we do?__

On the same run as adding the C8 element templates support, we have to make sure that we update [our docs](https://github.com/camunda/camunda-modeler/tree/develop/docs/element-templates). That includes

* Update screenshots
* Add Cloud samples
* Document the usage of the `$schema` attribute, mark Cloud templates 
* Add new / update features
  * `groups`
  * `optional` properties (only support `zeebe:input` and `zeebe:output`
  *  "new" entries visible = true
  *  Cloud bindings
    * `zeebe:taskDefinition`
    * `zeebe:taskHeader`
    * `zeebe:input`
    * `zeebe:output `
* Split up binding docs between Platform and Cloud (cf. https://github.com/camunda/camunda-modeler/issues/2382)
* Mark error scopes as Platform only

__Why should we do it?__

Keep things up to date
Child of #2640

### PR DESCRIPTION
**Pull request by pinussilvestrus**
Closes #2723 
Built on top #2785 

-----

* updates docs
  * update `$schema` attribute, required for Camunda Cloud 
  * update latest supported `$schema` versions
  * split up binding docs
  * add Camunda Cloud bindings
  * add optional bindings
  * add groups
  * add entriesVisible
* update samples
* update screenshots

### CODE REVIEW

**Comment by pinussilvestrus** [comment]
@MaxTru all the necessary changes should be included. Although it's waiting for #2785 to be merged, feel free to already have a look.

File: resources/element-templates/samples.json [hunk 1]
```diff
@@ -79,7 +79,106 @@
     ]
   },
   {
-    "$schema": "https://unpkg.com/@camunda/element-templates-json-schema@0.4.0/resources/schema.json",
+    "$schema": "https://unpkg.com/@camunda/element-templates-json-schema@0.6.0/resources/schema.json",
```
Comment (MaxTru) [inline resources/element-templates/samples.json:82]:
```suggestion
    "$schema": "https://unpkg.com/@camunda/element-templates-json-schema@0.7.0/resources/schema.json",
```

File: docs/element-templates/README.md [hunk 2]
```diff
@@ -122,15 +136,20 @@ As seen in the code snippet a template consist of a number of important componen
 
 ### JSON Schema Compatibility
 
-The application uses the `$schema` property to ensure compatibility for a given element template. The currently supported [Camunda element templates JSON Schema version](https://github.com/camunda/element-templates-json-schema) is `v0.4`. The Camunda Modeler will ignore element templates defining a higher `$schema` version. Although, it will log a warning message.
+The application uses the `$schema` property to ensure compatibility for a given element template. The latest supported [Camunda element templates JSON Schema versions](https://github.com/camunda/element-templates-json-schema) are
+
+* `v0.7.0` (Camunda Platform)
+* `v0.1.0` (Camunda Cloud)
+
+The Camunda Modeler will ignore element templates defining a higher `$schema` version. Although, it will log a warning message.
```
Comment (MaxTru) [inline docs/element-templates/README.md:144]:
```suggestion
The Camunda Modeler will ignore element templates defining a higher `$schema` version logging a warning message.
```

File: docs/element-templates/README.md [hunk 3]
```diff
@@ -309,78 +329,167 @@ Notice that adherence to the following configuration options is enforced by desi
 | **Binding parameters**  | `name`: the name of the property  |
 | **Mapping result** | `<... [name]=[userInput] ... />`  |
 
-##### `camunda:property`
+The `property` binding is supported both in Camunda Platform and Cloud.
```
Comment (MaxTru) [inline docs/element-templates/README.md:332]:
```suggestion
The `property` binding is supported both in Camunda Platform and Cloud.

##### Camunda Platform Bindings
```

**Review by MaxTru — COMMENTED** [review]
Great job! Only minor comments

File: docs/element-templates/README.md (hunk 3 above)
Comment (MaxTru) [inline docs/element-templates/README.md:332]: Also, for this binding I would add a heading (as below): `Bindings working for both, Camunda Platform and Camunda Cloud`

Comment (pinussilvestrus) [reply docs/element-templates/README.md:332]: Maybe `Product Agnostic Bindings`?

Comment (MaxTru) [reply docs/element-templates/README.md:332]: I like to have the explicit support in the headings. What about

```
Bindings for `Camunda Platform` or `Camunda Cloud`
Bindings for `Camunda Platform`
Bindings for `Camunda Cloud`
```

File: docs/element-templates/README.md (hunk 2 above)
Comment (pinussilvestrus) [reply docs/element-templates/README.md:144]: Or

```markdown
The Camunda Modeler will ignore element templates defining a higher `$schema` version and will log a warning message.
```

File: docs/element-templates/README.md [hunk 4]
```diff
@@ -309,78 +329,167 @@ Notice that adherence to the following configuration options is enforced by desi
...
+As of Camunda Modeler `v5.0.0`, we support optional bindings that do not persist empty values in the underlying BPMN 2.0 XML. 
+
+If a user removes the value in the configured control, it will also remove the mapped element.
+
+```json
+[
+  {
+    "$schema": "https://unpkg.com/@camunda/zeebe-element-templates-json-schema/resources/schema.json",
+    "name": "Task example",
+    "id": "some template",
```
[the remark this answers is not in the document]
Comment (pinussilvestrus) [reply docs/element-templates/README.md]: Good catch 👍

**Comment by pinussilvestrus** [comment]
Rebased on `develop` and adjusted the things you mentioned @MaxTru.

**Review by MaxTru — APPROVED** [review]
:clap:

**Pull request merged**